syntax = "proto3";

package todoPackage;

message Empty {};

message TodoItem {;
  int32 id = 1;
  string text = 2;
}

message TodoItems {
  repeated TodoItem items = 1;
}

service Todo {
  rpc createTodo(TodoItem) returns (TodoItem);
  rpc readTodos(Empty) returns (TodoItems);
  rpc readTodosStream(Empty) returns (stream TodoItem);
}


